Lib/json/encoder.py

```diff
@@ -244,10 +244,13 @@ def floatstr(o, allow_nan=self.allow_nan,
             return text
 
 
-        if (_one_shot and c_make_encoder is not None
-                and self.indent is None):
+        if _one_shot and c_make_encoder is not None:
+            if self.indent is not None and not isinstance(self.indent, str):
```

Comment:
We can simplify this condition:

```suggestion
            if isinstance(self.indent, int):
```

Reply:
Your suggestion is good, but there might be a few corner cases. What about values for `self.indent`that are not an instance of `int`, but do support `indent = ' ' * self.indent`?

I am ok with making the change, but would like the opinion of a core developer on this.

Reply:
You can also negate the condition:

```python
if self.indent is None or isinstance(self.indent, str):
    indent = self.indent
else:
    indent = ' ' * self.indent
```

Reply:
You can resolve this.